test/events.js

```diff
@@ -0,0 +1,222 @@
+'use strict';
+
+var Lab = require('lab');
+var lab = exports.lab = Lab.script();
+var describe = lab.describe;
+var it = lab.it;
+var before = lab.before;
+var beforeEach = lab.beforeEach;
+var after = lab.after;
+var afterEach = lab.afterEach;
+var Code = require('code');
+var expect = Code.expect;
+var sinon = require('sinon');
+var noop = require('101/noop');
+var os = require('os');
+var debug = require('debug');
+var EventEmitter = require('events').EventEmitter;
+
+require('loadenv')('cluster-man');
+var ClusterManager = require('../index.js');
+
+describe('cluster-man', function () {
+  describe('events', function() {
+    var manager;
+    var infoSpy;
+    var numWorkers = 4;
+
+    beforeEach(function (done) {
+      manager = new ClusterManager({
+        worker: noop,
+        numWorkers: numWorkers
+      });
+      var workerId = 0;
+      sinon.stub(manager.cluster, 'fork', function() {
+        return { id: ++workerId };
+      });
+      infoSpy = sinon.spy(manager.log, 'info');
+
+      // We are adding quite a few listeners over and over again
+      // since we instantiate and start the cluster manager throughout
+      // the tests. This will get rid of the "omg memory leak" warning.
```

Comment:
will we start getting this warning when, lets say, 10000 workers die in a non test environment?